src/interface/metadata.cpp

```diff
@@ -79,12 +83,19 @@ class UserMetadataState {
 
 parthenon::internal::UserMetadataState metadata_state;
 
-MetadataFlag Metadata::AllocateNewFlag(std::string &&name) {
-  return metadata_state.AllocateNewFlag(std::move(name));
+MetadataFlag Metadata::AllocateNewFlag(const std::string &name) {
```

Comment:
```suggestion
MetadataFlag Metadata::AddUserFlag(const std::string &name) {
```
In keeping with below.  Never liked `Allocate`, this isn't `malloc`

Reply:
Is anyone but you using this? That would be a breaking change. One I'm certainly fine with though.

Reply:
Doesn't changing all the `PackVariables` signatures count as a breaking change already?

Anyway, I don't think anyone else uses this, no -- @pdmullen mentioned looking around for uses and only seeing my `Params` craziness -- I certainly hope no one else bothered.

Reply:
The `PackVariables` signature is actually unchanged to some degree. The old signature still works because of C++ automatic casting.

Reply:
As in, passing an explicitly constructed `vector<MetadataFlag>{Metadata::Thing, UserMetadataFlag}` to the new functions will transparently construct a `FlagCollection`?  Neat.  Since I use the boolean on the end of `GetNames`, I kind of flipped the "this breaks things" switch and went wild.  As you know, Parthenon changes which are really not part of the interface tend to break things in KHARMA for hacky reasons :)

Reply:
> As in, passing an explicitly constructed vector<MetadataFlag>{Metadata::Thing, UserMetadataFlag} to the new functions will transparently construct a FlagCollection?

Yep :)

Reply:
Since you're the only one using this @bprather I am fine with changing the interface. Renaming to the suggested name.